Message-ID: <496AFF47.3010100@suse.de>
Date:	Mon, 12 Jan 2009 09:28:55 +0100
From:	Hannes Reinecke <hare@suse.de>
User-Agent: Thunderbird 2.0.0.12 (X11/20080226)
MIME-Version: 1.0
To:	James Bottomley <James.Bottomley@HansenPartnership.com>
Cc:	linux-scsi@vger.kernel.org
Subject: Re: [PATCH] Check for deleted device in scsi_device_online()
References: <20081216153352.27D4318C75E@pentland.suse.de> <1231432056.3237.25.camel@localhost.localdomain>
In-Reply-To: <1231432056.3237.25.camel@localhost.localdomain>
X-Enigmail-Version: 0.95.6
Sender:	linux-scsi-owner@vger.kernel.org
Precedence: bulk
List-ID: <linux-scsi.vger.kernel.org>
X-Mailing-List:	linux-scsi@vger.kernel.org


scsi_device_online() is not just a negation of SDEV_OFFLINE,
also devices in state SDEV_DEL are actually offline.

Signed-off-by: Hannes Reinecke <hare@suse.de>

diff --git a/include/scsi/scsi_device.h b/include/scsi/scsi_device.h
index e96daf6..bb80937 100644
--- a/include/scsi/scsi_device.h
+++ b/include/scsi/scsi_device.h
@@ -386,7 +386,8 @@ #define scmd_channel(scmd) sdev_channel((scmd)->device)
 
 static inline int scsi_device_online(struct scsi_device *sdev)
 {
-	return sdev->sdev_state != SDEV_OFFLINE;
+	return (sdev->sdev_state != SDEV_OFFLINE &&
+		sdev->sdev_state != SDEV_DEL);
 }
 
 /* accessor functions for the SCSI parameters */

-- 
Dr. Hannes Reinecke		      zSeries & Storage
hare@suse.de			      +49 911 74053 688
SUSE LINUX Products GmbH, Maxfeldstr. 5, 90409 Nürnberg
GF: Markus Rex, HRB 16746 (AG Nürnberg)
